server/routes/api/integrations.ts

```diff
@@ -43,20 +80,30 @@ router.post("integrations.update", auth(), async (ctx) => {
 
   assertArray(events, "events must be an array");
 
-  if (integration.type === "post") {
-    integration.events = events.filter((event: string) =>
-      ["documents.update", "documents.publish"].includes(event)
-    );
+  if (has(settings, "url")) {
+    assertUrl(settings.url);
   }
 
+  if (integration.type === IntegrationType.Post) {
+    integration.set({
+      events: events.filter((event: string) =>
+        ["documents.update", "documents.publish"].includes(event)
+      ),
+    });
+  }
+
+  integration.set({
+    settings,
+  });
```

Comment:
```suggestion
  integration.settings = settings;
```

Haven't used this `set` api before, seems unneccessary? :P

Reply:
Yes! Makes more sense when multiple attributes need to be set.

Reply:
Fixed!